### PR TITLE
caller to preprocess must be None aware

### LINKED ISSUE #33650: Empty vars_prompt causes exception 'NoneType' object is not iterable
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

vars_prompt

##### ANSIBLE VERSION

```
ansible 2.5.0 (devel 65ffb92bff) last updated 2017/12/06 20:34:58 (GMT +000)
  config file = /home/ansible/plays/ansible.cfg
  configured module search path = [u'/home/ansible/extensions/modules']
  ansible python module location = /home/ansible/src/ansible/lib/ansible
  executable location = /home/ansible/src/ansible/bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]

```

##### SUMMARY

Had a case where I commented out the variables in a vars_prompt section, but skipped the header and realized this caused an unfriendly NoneType exception...

##### STEPS TO REPRODUCE

```yaml
---

- hosts: localhost
  gather_facts: no
  connection: local

  vars_prompt:
#    - name: "some_prompt"
#      prompt: "Enter It"

  tasks:
  - name: version
    ios_command:
      commands: show version
```

##### EXPECTED RESULTS

A User-friendly error message or warning "Encountered an empty vars_prompt section in processing playbook"

##### ACTUAL RESULTS

```
Traceback (most recent call last):
  File "/home/ansible/src/ansible/bin/ansible-playbook", line 109, in <module>
    exit_code = cli.run()
  File "/home/ansible/src/ansible/lib/ansible/cli/playbook.py", line 130, in run
    results = pbex.run()
  File "/home/ansible/src/ansible/lib/ansible/executor/playbook_executor.py", line 82, in run
    pb = Playbook.load(playbook_path, variable_manager=self._variable_manager, loader=self._loader)
  File "/home/ansible/src/ansible/lib/ansible/playbook/__init__.py", line 54, in load
    pb._load_playbook_data(file_name=file_name, variable_manager=variable_manager)
  File "/home/ansible/src/ansible/lib/ansible/playbook/__init__.py", line 106, in _load_playbook_data
    entry_obj = Play.load(entry, variable_manager=variable_manager, loader=self._loader)
  File "/home/ansible/src/ansible/lib/ansible/playbook/play.py", line 112, in load
    return p.load_data(data, variable_manager=variable_manager, loader=loader)
  File "/home/ansible/src/ansible/lib/ansible/playbook/base.py", line 252, in load_data
    self._attributes[target_name] = method(name, ds[name])
  File "/home/ansible/src/ansible/lib/ansible/playbook/play.py", line 199, in _load_vars_prompt
    for prompt_data in new_ds:
TypeError: 'NoneType' object is not iterable
```

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY
also removed unused copy of method

fixes #33650

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

vars manager

##### ANSIBLE VERSION

```
2.4/2.5
```

### CODE REVIEW

File: lib/ansible/vars/manager.py [hunk 1]
```diff
@@ -151,26 +151,6 @@ def options_vars(self, value):
             raise AnsibleAssertionError("the type of 'value' for options_vars should be a dict, but is a %s" % type(value))
         self._options_vars = value.copy()
 
-    def _preprocess_vars(self, a):
```
Comment (maxamillion) [inline lib/ansible/vars/manager.py:154]: Is this just simply no longer needed? Doesn't seem related the the "None aware" change above so I wanted to make sure I wasn't missing something.

Comment (bcoca) [reply lib/ansible/vars/manager.py:154]: not needed, the code changed above called it at one point but doesn't anymore
